src/gentropy/dataset/study_locus.py

```diff
@@ -323,14 +323,14 @@ def _align_overlapping_tags(
     def assign_study_locus_id(
         study_id_col: Column,
         variant_id_col: Column,
-        finemapping_col: Column,
+        finemapping_col: Column = None,
     ) -> Column:
         """Hashes a column with a variant ID and a study ID to extract a consistent studyLocusId.
 
```

Comment:
Why we are not doing this:

```python
    @staticmethod
    def assing_identifier(*args: Column) -> Column:
        return f.xxhash64(args)
```

Consideration:
- This function would be in `Datasets` class and would be inherited to all datasets, because quite likely we'll need to generate some sort of identifiers.
- This approach would allow to arbitrarily increase the complexity of the hash generation. Eg. OK now we need a few more columns to generate the id. And we wouldn't need to touch the code anywhere else.
- I also don't like the idea to fake identifiers for rows where the variantId is missing. This problem should be sorted out when we are validating the data. row 358


Also why `finemapping_col: Column = None,` and not `finemapping_col: Column = f.lit(None)` then you would not need to test and write for the edge case, where finemapping column is not provided.

Reply:
Ah ok, I agree that solution is much more elegant and flexible. 

I also didn't realise you could use `Column = f.lit(None)` as a default value. Thanks. 

What should I do with this PR?

Reply:
Nothing, no action is required. It's just a potential abstraction.